File: apps/web/src/shared/api/provenance.ts
import type { ProvenanceMetadata as ApiProvenanceMetadata } from "@/src/shared/api/contract";

export interface ProvenanceRecord {
  framework?: string | null;
  frameworkVersion?: string | null;
  publishedAgentSnapshot?: Record<string, unknown> | null;
  artifactRef?: string | null;
  imageRef?: string | null;
  runnerBackend?: string | null;
  executorBackend?: string | null;
  traceBackend?: string | null;
  experimentId?: string | null;
  datasetVersionId?: string | null;
  datasetSampleId?: string | null;
}

export function mapProvenance(provenance?: ApiProvenanceMetadata | null): ProvenanceRecord | null {
  if (!provenance) {
    return null;
  }

  return {
    framework: provenance.framework ?? null,
    frameworkVersion: provenance.framework_version ?? null,
    publishedAgentSnapshot: provenance.published_agent_snapshot ?? null,
    artifactRef: provenance.artifact_ref ?? null,
    imageRef: provenance.image_ref ?? null,
    runnerBackend: provenance.runner_backend ?? null,
    executorBackend: provenance.executor_backend ?? null,
    traceBackend: provenance.trace_backend ?? null,
    experimentId: provenance.experiment_id ?? null,
    datasetVersionId: provenance.dataset_version_id ?? null,
    datasetSampleId: provenance.dataset_sample_id ?? null
  };
}

File: apps/control-plane/app/modules/shared/domain/provenance.py
from __future__ import annotations

from typing import Any
from uuid import UUID

from agent_atlas_contracts.execution import ExecutionTarget
from pydantic import BaseModel

from app.modules.shared.domain.execution import (
    EvaluatorConfig,
    ExecutionProfile,
    ToolsetConfig,
)
from app.modules.shared.domain.policies import ApprovalPolicySnapshot


class ProvenanceMetadata(BaseModel):
    agent_family: str | None = None
    framework: str | None = None
    framework_version: str | None = None
    published_agent_snapshot: dict[str, Any] | None = None
    execution_target: ExecutionTarget | None = None
    artifact_ref: str | None = None
    image_ref: str | None = None
    runner_backend: str | None = None
    executor_backend: str | None = None
    trace_backend: str | None = None
    experiment_id: UUID | None = None
    dataset_version_id: UUID | None = None
    dataset_sample_id: str | None = None
    approval_policy: ApprovalPolicySnapshot | None = None
    toolset: ToolsetConfig | None = None
    evaluator: EvaluatorConfig | None = None
    executor: ExecutionProfile | None = None
